pkg/integrations/statsd_exporter/metrics.go

```diff
@@ -0,0 +1,135 @@
+package statsd_exporter //nolint:golint
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// Metrics holds metrics used by the statsd_exporter integration. These metrics
+// are distinct from the set of metrics used by the exporter itself, and are just
+// used to monitor the stats of the listeners that forward data to the exporter.
+type Metrics struct {
+	EventStats            *prometheus.CounterVec
+	EventsFlushed         prometheus.Counter
+	EventsUnmapped        prometheus.Counter
+	UDPPackets            prometheus.Counter
+	TCPConnections        prometheus.Counter
+	TCPErrors             prometheus.Counter
+	TCPLineTooLong        prometheus.Counter
+	UnixgramPackets       prometheus.Counter
+	LinesReceived         prometheus.Counter
+	SamplesReceived       prometheus.Counter
+	SampleErrors          *prometheus.CounterVec
+	TagsReceived          prometheus.Counter
+	TagErrors             prometheus.Counter
+	MappingsCount         prometheus.Gauge
+	ConflictingEventStats *prometheus.CounterVec
+	ErrorEventStats       *prometheus.CounterVec
+	EventsActions         *prometheus.CounterVec
+	MetricsCount          *prometheus.GaugeVec
+}
+
+// NewMetrics initializes Metrics and registers them to the given Registerer.
+func NewMetrics(r prometheus.Registerer) (*Metrics, error) {
+	var m Metrics
+
+	m.EventStats = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "statsd_exporter_events_total",
+		Help: "The total number of StatsD events seen.",
+	}, []string{"type"})
+	m.EventsFlushed = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "statsd_exporter_event_queue_flushed_total",
+		Help: "Number of times events were flushed to exporter",
+	})
+	m.EventsUnmapped = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "statsd_exporter_events_unmapped_total",
```

Comment:
optional nit: I got told that best practice was to put the unit of measurement to the end of the metric, like `statsd_exporter_unmapped_events_total`

I don't have strong opinions about whether you do this.

Reply:
I would change it, but all of these metrics are copied straight out of statsd_exporter's main package, so I don't want to drift from the metrics name it uses.

Reply:
That makes sense to me then.